src/Foundation/NSObject2.cs

```diff
@@ -155,6 +161,34 @@ internal static IntPtr CreateNSObject (IntPtr type_gchandle, IntPtr handle, Flag
 			return Runtime.AllocGCHandle (obj);
 		}
 
+		IntPtr GetSuper ()
+		{
+			if (super == IntPtr.Zero) {
+				IntPtr ptr;
+
+				unsafe {
+					ptr = Marshal.AllocHGlobal (sizeof (objc_super));
+					*(objc_super*) ptr = default (objc_super); // zero fill
+				}
+
+				var previousValue = Interlocked.CompareExchange (ref super, ptr, IntPtr.Zero);
+				if (previousValue != IntPtr.Zero) {
+					// somebody beat us to the assignment.
+					Marshal.FreeHGlobal (ptr);
+					ptr = IntPtr.Zero;
+				}
```

Comment:
AFAIK, same, not really need to check against IntPtr.Zeo

Reply:
Yes, here we do, because we're not comparing and freeing the same pointer.